### PR TITLE
feat(build): implement build command

### PR DESCRIPTION
**Pull request by BeniCheni**
Refer to [tink-build topic](https://npm.community/t/tink-implement-tink-build/3313).

- [x] Implementation
- [x] Add unit test

###### Without `build` run-script
No script is run.
```sh
→ tink build
```

###### With a dummy script, i.e. `"build": "echo Bingo!"`
`build` run-script is executed to print `Bingo!`.
```sh
→ tink build

> tink@0.14.0 build /Users/bchen/github/tink
> echo Bingo!

Bingo!
```

### CODE REVIEW

File: bin/tink.js [hunk 1]
```diff
@@ -4,12 +4,13 @@ require('../lib/node/index.js')
...
-  ['rm', require('../lib/commands/rm.js')],
-  ['shell', require('../lib/commands/shell.js')],
+  ['build', require('../lib/commands/build.js')],
+  ['deprecate', require('../lib/commands/deprecate.js')],
   ['org', require('../lib/commands/org.jsx')],
   ['prepare', require('../lib/commands/prepare.js')],
   ['ping', require('../lib/commands/ping.js')],
-  ['deprecate', require('../lib/commands/deprecate.js')]
+  ['rm', require('../lib/commands/rm.js')],
+  ['shell', require('../lib/commands/shell.js')]
```
Comment (BeniCheni) [inline bin/tink.js]: Minor FYI: ☝️while adding the required `build.js`, why not sort these items alphabetically.
(Happy to revert sort if it's bad idea.)

File: lib/commands/build.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    nodeOptions: { default: {} },
+    config: { default: {} },
+    unsafePerm: { default: false },
+    scriptsPrependNodePath: { default: false },
+    ignoreScripts: { default: false },
+    user: {},
+    group: {},
+    stdio: {},
+    scriptShell: {}
+  })
```
Comment (BeniCheni) [inline lib/commands/build.js:34]: Minor FYI: these default values of the `BuildConfig` are based on my best guesses, while reading code about how these option keys are being used in [npm-lifecycle/index.js](https://github.com/npm/npm-lifecycle/blob/latest/index.js).

Please advise if any of these default values, or the way of how this config object is written, can be improved. 🙏

File: lib/commands/build.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  builder (y) {
+    return y.help().alias('help', 'h')
+      .options(Build.options)
+  },
+  options: Object.assign(require('../common-opts.js', {})),
+  handler: async argv => build(argv)
+}
+
+async function build (argv) {
+  const findPrefix = require('find-npm-prefix')
```
Comment (zkat) [inline lib/commands/build.js]: This is available as `libnpm.getPrefix()`

File: lib/commands/build.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  const readJson = require('read-package-json')
```
Comment (zkat) [inline lib/commands/build.js]: This one is available as `libnpm.readJSON()`

File: lib/commands/build.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  findPrefix(process.cwd()).then(prefix => {
+    const packageJson = `${prefix}/package.json`
+    readJson(
+      packageJson,
+      console.error,
+      false,
+      async (er, data) => {
+        if (er) {
+          console.error(`There was an error reading ${packageJson}`)
```
Comment (zkat) [inline lib/commands/build.js]: `libnpm.log.error()` for this and the one below, please.

**Review by zkat — CHANGES_REQUESTED** [review]
Just a few small nits so far. Looking forward to having tests for it! 💚 Everything looks great and I appreciate the contribution!

**Comment by BeniCheni** [comment]
Hi @zkat. Updated per the great nits you pointed out, updated the source to better support testability & wrote a basic test to assert `true` to validate the successful execution of the `build` command. Hopefully, this `build` PR is in good shape for review again for a wrap. Please review at convenience.

###### Trivial Observation
> File                |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s 
> build.js          |    93.33 |      100 |      100 |    93.33 |                 7 

The uncovered line 7 is
```js
return y.help().alias('help', 'h')
```
which are also observed for the `deprecate.js` & `ping.js` tests. So wondering if this uncovered line observation is rather "_universal_" among some tests. So I left it as it is for now, as % coverage threshold has not been set up yet.

**Comment by BeniCheni** [comment]
Ping/CC @zkat. Friendly FYI the branch of this PR has been rebased with `latest`, after the recent merge of other PRs. This PR should be in good shape for final review.

File: lib/commands/build.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    user: {},
+    group: {},
+    stdio: {},
+    scriptShell: {}
+  })
+  const opts = BuildConfig(npmConfig().concat(argv).concat({ log }))
+  return await libnpm.getPrefix(process.cwd())
+    .then(prefix => {
+      const packageJSON = `${prefix}/package.json`
+      libnpm.readJSON(packageJSON)
```
Comment (zkat) [inline lib/commands/build.js]: You should use an `await` here as well and get rid of the .then()s

**Comment by BeniCheni** [comment]
@zkat, indeed, I was able to improve the code to adapt `await` syntax to omit the `.then` usage in [this commit](https://github.com/npm/tink/pull/29/commits/94ec2a62bab2d8ffe6f3a409070e0309f222d75d). Please take a quick look at convenience. Thanks!

**Review by zkat — APPROVED** [review]

**Pull request merged**